tests/client/test_async_client.py

```diff
@@ -146,3 +146,13 @@
 
     assert response.status_code == 200
     assert response.content == data
+
+
+async def test_async_client_timeout_default(server, backend):
+    async with httpx.AsyncClient(backend=backend) as client:
+        assert client.dispatch.timeout == httpx.config.UNSET
```

Comment:
So, maybe this is an odd behavior. Shouldn't we expect accessing the timeout config at any point in the chain to return a definite `TimeoutConfig` object? cc @tomchristie